by-language/java-jdbc/BasicPostgresCrateDB.java

```diff
@@ -5,50 +5,70 @@
 import java.sql.ResultSetMetaData;
 import java.util.Locale;
 import java.util.Properties;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.JCommander;
 
 public class BasicPostgresCrateDB {
 
+    @Parameter(names = {"--dburl", "-d"}, description = "Database URL")
+    String dburl;
+    @Parameter(names = {"--user", "-u"}, description = "Username")
+    String user = "crate";
+    @Parameter(names = {"--password", "-p"}, description = "Password")
+    String password = "";
+
+    @Parameter(names = "--help", help = true, description = "Display this page")
+    private boolean help;
+
     public static final int SO_RCVBUF = 1024 * 1024; // 1 MB
-    public static final int MAX_BATCH_SIZE = 20000;
+    public static final int MAX_BATCH_SIZE = 10000;
     public static final int QUERY_EXECUTION_TIMEOUT_SECS = 60;
 
+    public static void main(String ... argv) throws Exception {
 
-    public static void main(String[] args) throws Exception {
-
-        // Create schema and insert data:
-        //
-        // psql postgres://crate@localhost:5432/
-        //
-        // CREATE TABLE testdrive (id INT PRIMARY KEY, data TEXT);
-        // INSERT INTO testdrive VALUES (0, 'zero'), (1, 'one'), (2, 'two');
+        BasicPostgresCrateDB main = new BasicPostgresCrateDB();
+        JCommander jc = JCommander.newBuilder()
+                .addObject(main)
+                .build();
+        jc.setProgramName("cratedb-jdbc-example");
+        jc.parse(argv);
+        int exitcode = main.run(jc);
+        System.exit(exitcode);
+    }
+    public int run(JCommander jc) throws Exception {
+        /*
+         * Connect to CrateDB and query `sys.summits` table.
+         * https://jdbc.postgresql.org/documentation/head/connect.html
+         */
+        if (help) {
+            jc.usage();
+            return 0;
+        }
 
-        // String connectionUrl = "jdbc:postgresql://localhost:5432/";
-        // String connectionUrl = "jdbc:crate://localhost:5432/";
-        String connectionUrl = args[0];
+        String connectionUrl = dburl;
 
         Properties connectionProps = new Properties();
-        // https://jdbc.postgresql.org/documentation/head/connect.html
-        connectionProps.put("user", "crate");
-        connectionProps.put("password", "");
-        connectionProps.put("ssl", false);
+        connectionProps.put("user", user);
+        connectionProps.put("password", password);
+        //connectionProps.put("ssl", "true");
```

Comment:
Apparently, setting the `ssl` parameter is not needed. The driver itself seems to figure out how to correctly connect to both a non-SSL protected instance on localhost, and an SSL-protected instance on CrateDB Cloud, without needing to configure it.